nixos/modules/services/taler/exchange.nix

```diff
@@ -0,0 +1,184 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+
+let
+  this = config.services.taler.exchange;
+  # Services that need access to the DB
+  servicesDB = [
+    "httpd"
+    "aggregator"
+    "closer"
+    "wirewatch"
+  ];
+  # Services that do not need access to the DB
+  servicesNoDB = [
+    "secmod-cs"
+    "secmod-eddsa"
+    "secmod-rsa"
+  ];
+  services = servicesDB ++ servicesNoDB;
+  dbName = "taler-exchange-httpd";
+  # taler-exchange needs a runtime dir shared between the taler services. Crypto
+  # helpers put their sockets here for instance and the httpd connects to them.
+  runtimeDir = "/run/taler-system-runtime/";
+  talerEnabled = config.services.taler.enable;
+  inherit (config.services.taler) configFile;
+in
+
+{
+  options.services.taler.exchange = {
+    enable = lib.mkEnableOption "the GNU Taler exchange";
+    package = lib.mkPackageOption pkgs "taler-exchange" { };
+    denominationConfig = lib.mkOption {
+      type = lib.types.lines;
+      default = throw "You must set the denomination config `services.taler.exchange.denominationConfig`.";
+      example = ''
+        [COIN-KUDOS-n1-t1718140083]
+        VALUE = KUDOS:0.1
+        DURATION_WITHDRAW = 7 days
+        DURATION_SPEND = 2 years
+        DURATION_LEGAL = 6 years
+        FEE_WITHDRAW = KUDOS:0
+        FEE_DEPOSIT = KUDOS:0.1
+        FEE_REFRESH = KUDOS:0
+        FEE_REFUND = KUDOS:0
+        RSA_KEYSIZE = 2048
+        CIPHER = RSA
+      '';
+      description = ''
+        This option configures the cash denomination for the coins that the exchange offers.
+        For more information, consult the docs: https://docs.taler.net/taler-exchange-manual.html#coins-denomination-keys
+
+        You can either write these manually or you can use the `taler-wallet-cli deployment gen-coin-config`
+        command to generate it.
+
+        Warning: Do not modify existing denominations after deployment.
+        Please see the upstream docs for how to safely do that.
+      '';
+    };
+    debug = lib.mkEnableOption "debug logging";
+  };
+
+  config = lib.mkIf (talerEnabled && this.enable) {
+    services.taler.includes = [
+      (pkgs.writers.writeText "exchange-denominations.conf" this.denominationConfig)
+    ];
+
+    systemd.slices.taler-exchange = {
+      description = "Slice for GNU taler exchange processes";
+      before = [ "slices.target" ];
+    };
+
+    systemd.services =
+      lib.genAttrs (map (n: "taler-exchange-${n}") services) (name: {
+        serviceConfig = {
+          DynamicUser = true;
+          User = name;
+          Group = "taler-exchange"; # TODO refactor into constant
+          ExecStart =
+            "${this.package}/bin/${name} -c ${configFile}" + lib.optionalString this.debug " -L debug"; # TODO as a list?
```

Comment:
Please use lib.getExe. Also after ExecStart should be no new line

Reply:
This is already the case ([here](https://github.com/NixOS/nixpkgs/pull/332699/files#diff-7c8ae3dbbd2b175f1e5d71e2fe0876f8e4d5b285c5681a2948d3ba5023d6b6c2R156)).

As for the new line, this is how nixfmt formats this. What do you suggest to do instead?